### PR TITLE
improvement: test file structure === server file structure

### PR DESCRIPTION
**Pull request by kirrg001**
no issue
- ensure we have the same structure in test folder (especially for unit tests)
- it's hard as contributor to look for tests, when they don't sit in the correct folder
- for now: we have some `index_spec.js` files because of the server structure
- future: do not use index_spec.js to integrate logic, this is not the purpose of an index.js
- created `extra` folder, because we had tests for `shared` folder and general tests which don't belong to unit or integration
### notes

Some Pull Requests needs a rebase with master after this change.
### coverage-all master

Statements   : 90.88% ( 6653/7321 )
Branches     : 85.05% ( 3368/3960 )
Functions    : 88.88% ( 1623/1826 )
Lines        : 90.94% ( 6643/7305 )
### coverage-all this PR

Statements   : 90.88% ( 6653/7321 )
Branches     : 85.05% ( 3369/3961 )
Functions    : 88.88% ( 1623/1826 )
Lines        : 90.94% ( 6643/7305 )
### tasks
- [x] wait that #6955 and #6929 are merged, they both cause random errors
- [x] recheck coverage
- [x] go over each test file and check if i deleted some new code
- [x] rename others

### CODE REVIEW

**Comment by acburdine** [comment]
@kirrg001 when you've got a bit, both of those PR's mentioned in tasks have been merged, so would this work after a rebase/

File: core/test/unit/apps_filters_spec.js [hunk 1]
```diff
@@ -1,145 +0,0 @@
-var should  = require('should'),
```
Comment (kirrg001) [inline core/test/unit/apps_filters_spec.js]: this file was a duplicate of `filters_spec.js`

File: core/test/unit/apps_spec.js [hunk 2]
```diff
@@ -1,526 +0,0 @@
-var path         = require('path'),
```
Comment (kirrg001) [inline core/test/unit/apps_spec.js]: this file i have created four different specs for proxy, permissions, sandbox and dependencies (see `core/test/unit/apps`.

**Comment by acburdine** [comment]
@ErisDS do you have any thoughts on this before it gets merged?

**Comment by ErisDS** [comment]
This is all much, much cleaner 👍 

Only thing I'm not 100% on is the others thing. Partly because that's an odd word, for showdown perhaps 3rd party or external is what's meant? Other reason is that these tests do contribute towards the coverage level, I think, and so it seems weird to separate them off?

**Comment by kirrg001** [comment]
will add to `coverage-all`, but not sure about the naming "external". maybe "extra"?

**Comment by acburdine** [comment]
"extra" sounds good to me 😄

**Comment by ErisDS** [comment]
Sorry for dropping this. I also agree that `extra` makes sense. Would you like to get this merged ?

**Comment by kirrg001** [comment]
I would close it :( 
It's too old now. Too many conflicts to resolve - especially for our `playground` branch. 
Will do it again in the future for `1.0`.

**Pull request closed without merging**